File: createcell.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

int cell_create(void){
	int current_slots = 0, wanted_slots, read_input;
	
	while (read_input != 0){
		cout << "there are currently " << current_slots << " slots in your inventory." << endl;
		cin << read_input;
		wanted_slots = read_input + current_slots;
		for (int i = current_slots + 1; i != wanted_slots; current_slots++){
			inventory.push_back({});
			current_slots = i;
		}
		cout << "Done!" << endl;
		cout << "Total slots: " << current_slots << endl;
		cout << "Wanted slots: " << wanted_slots << endl;
		wanted_slots = 0;
	}
	
}

int main(void){
	vector <vector<int>> inventory;
	cell_create();
	
	for (int i = 0; 0 != sizeof(inventory); ++i){
		cout << i
	}
	return 0;
}

File: createcell.lua
local inventory = {};
local current_slots = 0;
local wanted_slots = 0;
local read_input = -1;

while (read_input ~= 0) do
print("There are currently ".. current_slots .. " slots in your inventoy");
io.write("Create how many slots: ");
read_input = tonumber(io.read());
wanted_slots = read_input + current_slots; --because yes
for i = (current_slots+1), wanted_slots do
    inventory[i] = {};
    current_slots = i;

end
print("Total slots: "..current_slots)
print("Wanted slots: "..read_input);
wanted_slots = 0;
for k, v in ipairs(inventory) do
    print(k, v);
end
end

--[[function getItemAndCount()
    
end
function create_cell()
    
end
]]--